### DISCUSSION TITLE
Disabling Captcha for phone number authentication

### DISCUSSION THREAD

**Discussion by davebanguilan** [capawesome-team/capacitor-firebase#219] [Q&A]
I was able to add the SHA fingerprint both SHA-1 and SHA-256. But when I am testing it, the app opened chrome for a bit (It says verifying for Captcha) then it redirected back to my app. How can I avoid the Captcha part?

**Comment by robingenz** [comment]
Maybe this helps you: https://stackoverflow.com/a/65144360/6731412

> Also, need to perform additional steps: Firebase Project Settings > App check > and Register firebase project in SafetyNet and Play Integrity register with default time token 1 hour.

------------

I have not tested it myself yet. There is also an open issue for Firebase AppCheck: #177

**Comment by gelinger777** [comment]
@robingenz   we want to activate integrity check on the app . Do you maybe already have some small instruction how to update from old version to new one so that the new sdk will be used and what exactly we should do on Firebase/Google Cloud console this to happen? 

Is the step of adding 

```implementation 'androidx.browser:browser:1.2.0'  ```    needed? It seems very little info in docs and tickets in this regards with your library .

**Reply by robingenz** [reply]
> we want to activate integrity check on the app 

You mean you want to use Firebase App Check. Here you can find the docs: https://capawesome.io/plugins/firebase/app-check/

Please do not post unrelated replies to discussions and create a new discussion instead.

**Reply by gelinger777** [reply]
@robingenz  No I don't want to use app-check. I don't know how to make phone authentication to work.  because Safety Net checks are obsolete.  And now Phone Verification has to be done  other way.

**Reply by gelinger777** [reply]
@robingenz  https://github.com/capawesome-team/capacitor-firebase/issues/293#issuecomment-1488639163 here you say that it's just about SDK  and new version solves the problem , but it seems it is not.  

![image](https://github.com/capawesome-team/capacitor-firebase/assets/184276/83c732aa-ea40-493a-ad7f-8dc77af0dcff)

With updated  Package  we are getting this.

**Reply by robingenz** [reply]
> I don't know how to make phone authentication to work.

Nothing special here. Just follow the [setup guide](https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/authentication/docs/setup-phone.md).

We also provide an [example](https://capawesome.io/plugins/firebase/authentication/#usage):

```ts
const signInWithPhoneNumber = async () => {
  return new Promise(async resolve => {
    // Attach `phoneCodeSent` listener to be notified as soon as the SMS is sent
    await FirebaseAuthentication.addListener('phoneCodeSent', async event => {
      // Ask the user for the SMS code
      const verificationCode = window.prompt(
        'Please enter the verification code that was sent to your mobile device.',
      );
      // Confirm the verification code
      const result = await FirebaseAuthentication.confirmVerificationCode({
        verificationId: event.verificationId,
        verificationCode,
      });
      resolve(result.user);
    });
    // Attach `phoneVerificationCompleted` listener to be notified if phone verification could be finished automatically
    await FirebaseAuthentication.addListener(
      'phoneVerificationCompleted',
      async event => {
        resolve(event.result.user);
      },
    );
    // Start sign in with phone number and send the SMS
    await FirebaseAuthentication.signInWithPhoneNumber({
      phoneNumber: '123456789',
    });
  });
};
```

If this does not work, please create a bug report. The error can have different reasons, see this [SO thread](https://stackoverflow.com/questions/65490555/this-request-is-missing-a-valid-app-identifier-meaning-that-neither-safetynet-c).

**Reply by gelinger777** [reply]
Dear @robingenz  everything was working  prior the Integrity Check  problem. The most unclear part is  Step N4

```See [Enable app verification](https://firebase.google.com/docs/auth/android/phone-auth#enable-app-verification) and follow the instructions to enable app verification correctly.```

**Reply by robingenz** [reply]
Your problem is not directly related to the plugin, but to the configuration of the Firebase SDK. Please reach out to the Firebase team. I can only provide limited support for the plugin. Again, i recommend taking a look at this [SO thread](https://stackoverflow.com/questions/65490555/this-request-is-missing-a-valid-app-identifier-meaning-that-neither-safetynet-c).
